src/main/java/teammates/ui/webapi/GetFeedbackSessionLogsAction.java

```diff
@@ -126,51 +99,87 @@ public JsonResult execute() {
             throw new InvalidHttpParameterException("The end time should be after the start time.");
         }
 
-        long earliestSearchTime = TimeHelper.getInstantDaysOffsetBeforeNow(Const.LOGS_RETENTION_PERIOD.toDays())
-                .toEpochMilli();
-        if (startTime < earliestSearchTime) {
-            throw new InvalidHttpParameterException(
-                    "The earliest date you can search for is " + Const.LOGS_RETENTION_PERIOD.toDays() + " days before today."
-            );
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+
+        if (!isCourseMigrated(courseId)) {
+            long earliestSearchTime = TimeHelper.getInstantDaysOffsetBeforeNow(Const.LOGS_RETENTION_PERIOD.toDays())
+                    .toEpochMilli();
+            if (startTime < earliestSearchTime) {
+                throw new InvalidHttpParameterException("The earliest date you can search for is "
+                        + Const.LOGS_RETENTION_PERIOD.toDays() + " days before today.");
+            }
         }
 
-        List<FeedbackSessionLogEntry> fsLogEntries =
-                logsProcessor.getOrderedFeedbackSessionLogs(courseId, email, startTime, endTime, feedbackSessionName);
+        String email = getRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+        String feedbackSessionName = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
 
         if (isCourseMigrated(courseId)) {
+            if (sqlLogic.getCourse(courseId) == null) {
+                throw new EntityNotFoundException("Course not found");
+            }
+
+            if (email != null && sqlLogic.getStudentForEmail(courseId, email) == null) {
+                throw new EntityNotFoundException("Student not found");
+            }
+
+            if (feedbackSessionName != null && sqlLogic.getFeedbackSession(feedbackSessionName, courseId) == null) {
+                throw new EntityNotFoundException("Feedback session not found");
+            }
+        } else {
+            if (logic.getCourse(courseId) == null) {
+                throw new EntityNotFoundException("Course not found");
+            }
+
+            if (email != null && logic.getStudentForEmail(courseId, email) == null) {
+                throw new EntityNotFoundException("Student not found");
+            }
+
+            if (feedbackSessionName != null && logic.getFeedbackSession(feedbackSessionName, courseId) == null) {
+                throw new EntityNotFoundException("Feedback session not found");
+            }
```

Comment:
place this block in the else block, for future developers to remove it more easily